main/lsp/updates.cc

```diff
@@ -262,7 +262,10 @@ LSPLoop::TypecheckRun LSPLoop::tryFastPath(unique_ptr<core::GlobalState> gs,
                     finalGs = core::GlobalState::replaceFile(move(finalGs), fref, changedFiles[i]);
                     subset.emplace_back(fref);
                 }
-                globalStateHashes[fref.id()] = hashes[i];
+                // Ignore invalid hash states (caused by syntax errors).
+                if (newFile || hashes[i].definitions.hierarchyHash != core::GlobalStateHash::HASH_STATE_INVALID) {
```

Comment:
I think this can lead to situation where all other files have been typecheked under incorrect global state and they don't get invalidated.
I think we need to update hashing mechanism itself to not return `HASH_STATE_INVALID` for files that don't parse.

Reply:
Ah, I thought we discussed this as a viable solution in our 1:1. But now that I think of it...

1. LSP applies update with changes to files {A, B}. A has syntax error, B calls something in A incorrectly -- but typechecks anyway.
2. LSP applies update that fixes A. A retypechecks on fast path, but B doesn't -- so error is suppressed.